gdnative-core/src/export/property/hint.rs

```diff
@@ -116,12 +116,22 @@ where
 /// ```
 #[derive(Clone, Eq, PartialEq, Debug, Default)]
 pub struct EnumHint {
-    values: Vec<String>,
+    values: Vec<(String, Option<i64>)>,
 }
 
 impl EnumHint {
     #[inline]
     pub fn new(values: Vec<String>) -> Self {
+        let values = values.into_iter().map(|v| (v, None)).collect();
+        EnumHint { values }
+    }
+
+    #[inline]
+    pub fn with_numbers(values: Vec<(String, i64)>) -> Self {
+        let values = values
+            .into_iter()
+            .map(|(key, val)| (key, Some(val)))
+            .collect();
         EnumHint { values }
     }
```

Comment:
Is the "partially numbered" format ever useful to represent? We should expose a constructor for that format if Godot does indeed recognize it. Otherwise, I'd prefer a type definition that makes the situation clearer for the future, e.g. the following:

```rust
pub struct EnumHint {
    kind: EnumHintKind,
}

enum EnumHintKind {
    Implicit(Vec<String>),
    Explicit(Vec<(String, i64)>),
}
```

Reply:
I see. I guess that's a possibility, although even better I suppose would be to accept a `struct EnumHintEntry { name: String, value: Option<i64> }` to which further fields can be added in the future in case it ever becomes necessary for any reason. Whether it should be made `#[non_exhaustive]` with public fields or private with `with_*` constructors I'm not 100% certain, but personally I lean toward the latter.